cmd/kube-proxy/app/options/options.go

```diff
@@ -57,7 +57,6 @@ func NewProxyConfig() *ProxyServerConfig {
 			ResourceContainer:              "/kube-proxy",
 			IPTablesSyncPeriod:             unversioned.Duration{30 * time.Second},
 			UDPIdleTimeout:                 unversioned.Duration{250 * time.Millisecond},
-			Mode:                           componentconfig.ProxyModeUserspace,
```

Comment:
I think the reason to leave it unset is to allow that to mean "use what's best", which in this case translates to iptables. Also iptables auto downgrades to userspace if the wrong version of iptables is present. I agree, we need a clearer way to surface this. And a better way to test it.